codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -496,8 +518,20 @@ private void incrementPendingBytes(int numBytes) {
              */
             int write(int allowedBytes) {
                 int before = payload.size();
-                needFlush |= payload.write(Math.max(0, allowedBytes));
-                int writtenBytes = before - payload.size();
+                int writtenBytes = 0;
+                boolean success = false;
+                try {
+                    needFlush |= payload.write(Math.max(0, allowedBytes));
+                    success = true;
+                } catch (Throwable t) {
+                    payload.error(t);
+                } finally {
+                    if (success && payload.size() == 0) {
+                        pendingWriteQueue.remove();
+                        payload.writeComplete();
+                    }
+                }
+                writtenBytes = before - payload.size();
```

Comment:
@nmittler actually I think we don't need the finally block here at all? It doesn't do anything.